c10/core/TensorImpl.h

```diff
@@ -563,6 +563,11 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     key_set_ = key_set_ - autograd_dispatch_keyset;
   }
 
+  // Inference tensor doesn't have autograd or InplaceOrView key.
+  bool is_inference_tensor() {
+    return !key_set_.has(c10::DispatchKey::InplaceOrView) && (key_set_ & c10::autograd_dispatch_keyset).empty();
```

Comment:
Since we read both keys here, should we have a TORCH_INTERNAL_CHECK that makes sure that we don't have any invalid state where one is set but not the other?